app/views/contexts/summary/group.py

```diff
@@ -77,6 +83,11 @@ def _build_blocks(
                     ]
                 )
 
+            elif block["id"] in routing_path and block["type"] in ["ListCollector"]:
+                if summary_elements.get("custom_summary"):
+                    for list_element in summary_elements["custom_summary"]:
+                        blocks.extend([list_element])
```

Comment:
As discussed, the summary element shouldn't be passed here and then remapped, we should build what we need here. You are always given everything you need to build the list element. The list element should really be a new List collector block model but being pragmatic due to time constraints, so happy to keep this as a dict even though it muddies the types.

Reply:
Ok, new `ListCollectorBlock` class added and related methods moved to that class from `SectionSummaryContext`.